src/pgduckdb_types.cpp

```diff
@@ -337,6 +337,16 @@ ConvertNumeric(const duckdb::Value &ddb_value, idx_t scale, NumericVar &result)
 static Datum
 ConvertNumericDatum(const duckdb::Value &value) {
 	auto value_type_id = value.type().id();
+
+	// Special handle duckdb VARINT type.
+	if (value.type().id() == duckdb::LogicalTypeId::VARINT) {
+		std::string value_str = value.ToString();
+		Datum pg_numeric = DirectFunctionCall3(numeric_in, CStringGetDatum(value_str.c_str()),
+		                                       ObjectIdGetDatum(InvalidOid), Int32GetDatum(-1));
```

Comment:
For this conversion I appreciate the simplicity that the string conversion gives us. Especially since VARINTs are fairly rare types and already a pretty expensive anyway. So converting making the conversion to Postgres a bit more expensive than strictly necessary seems acceptable. This does need a `PostgresFunctionGuard` though, to keep Postgres C code a bit more physically separated from the C++ code let's move this to logic to the `src/pg/types.cpp` file. (we're steadily moving more code to the `src/pg` directory, but we're not done with that migration yet)

Reply:
> This does need a PostgresFunctionGuard though, to keep Postgres C code a bit more physically separated from the C++ code

Make sense! Function guard added.

Just to confirm, I don't think we need type conversion for numerics <-> VARINT, right?
Because the entrypoint is in pg and we have other mapping (double) in duckdb.

> let's move this to logic to the src/pg/types.cpp file.

I could understand the motivation (i.e. place pg related code under `pg` directory for better maintainability), but I'm not sure which part to place, should I move all functions there?

As you might be aware of, I'm also working on other type conversions; do you think it ok after I finish them?
I'm willing to help migration later on :)